game/script-farewells.rpy

```diff
@@ -2,42 +2,757 @@ default persistent._farewell_database = dict()
 
 init python in farewells:
     import random
+    import store
+
+    # Natsuki will not ask a player to stay again if they agreed previously
+    store.jn_globals.player_already_stayed = False
+
     FAREWELL_MAP = dict()
 
+    def get_farewell_in_affinity_range(farewell):
+        """
+        Returns true if player's persistent value is within the farewell topic's affinity range
+
+        IN:
+            farewell - The farewell topic to check
+        """
+        if store.persistent.affinity in range(farewell.affinity_range[0], farewell.affinity_range[1]):
+            return True
+            
+        else:
+            return False
```

Comment:
Can avoid the `if true: return true`
```suggestion
        return store.persistent.affinity in range(farewell.affinity_range[0], farewell.affinity_range[1])
```
Additionally, we can probably build this onto `Topic` to check this, especially if we'd be evaluating conditionals

So I'm thinking we add a staticmethod to `Topic` to check a Topic's affinity_range vs the current affinity, we can then make another function which iterates a list thru that function to filter it down instead.